test/e2e/util.go

```diff
@@ -505,6 +505,18 @@ func buildCatalogSourceCleanupFunc(crc versioned.Interface, namespace string, ca
 		ctx.Ctx().Logf("Deleting catalog source %s...", catalogSource.GetName())
 		err := crc.OperatorsV1alpha1().CatalogSources(namespace).Delete(context.Background(), catalogSource.GetName(), metav1.DeleteOptions{})
 		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func() (bool, error) {
+			fetched, err := newKubeClient().KubernetesInterface().CoreV1().Pods(catalogSource.GetNamespace()).List(context.Background(), metav1.ListOptions{LabelSelector: "olm.catalogSource=" + catalogSource.GetName()})
```

Comment:
Any thoughts on whether we should update the parent function signature and pass the kube client as a parameter vs. utilizing the helper function to get it at runtime? I'm not super I have a strong opinion, but wanted to get another opinion if we think passing it as a parameter is cleaner (or easier to maintain).

Reply:
I don't have a strong opinion, either.  I like this way because it makes this PR changes small.  I can implement passing the kube client as an argument.  It needs some additional changes in this file and one more file.  What do you think?

Reply:
Sounds reasonable to me - lets hold off.

/lgtm

Reply:
I think we should fix this holistically across the test suite, I've created [this issue](https://github.com/operator-framework/operator-lifecycle-manager/issues/2570) to track the work.